channeld/channel.c

```diff
@@ -1849,10 +1854,19 @@ static void peer_reconnect(struct peer *peer)
 	 *     of the next `commitment_signed` it expects to receive.
 	 *   - MUST set `next_remote_revocation_number` to the commitment number
 	 *     of the next `revoke_and_ack` message it expects to receive.
+	 *   - if it supports `option_data_loss_protect`:
+	 *     - if `next_remote_revocation_number` equals 0:
+	 *       - MUST set `your_last_per_commitment_secret` to all zeroes
+	 *     - otherwise:
+	 *       - MUST set `your_last_per_commitment_secret` to the last
+	 *         `per_commitment_secret` it received
 	 */
-	msg = towire_channel_reestablish(NULL, &peer->channel_id,
-					 peer->next_index[LOCAL],
-					 peer->revocations_received);
+	msg = towire_channel_reestablish_option_data_loss_protect
+		(NULL, &peer->channel_id,
```

Comment:
Nit: code style seems off here, the `(` should be part of the previous line shouldn't it?

Reply:
It's hard with names this long; I have no strong opinion either way since it's so ugly already.

But usually I prefer operators on the line below, as it makes it visually obvious that it's a continuation, eg:

```
    x = really_long thing
         + something else;
```